_stbt/control_gpl.py

```diff
@@ -217,16 +217,23 @@ def keyup(self, key):
         debug("Released %s" % key)
 
     def send_keydowns(self, key):
+        # CEC spec section 13.13.3 says that the receiver should assume a keyup
+        # if it hasn't seen one within a receiver-determined timeframe which
+        # can't be less than 550ms. For press-and-hold the initiator should
+        # send repeated keydown commands (200 to 450ms apart) followed by the
+        # final keyup.
         end_time = time.time() + 60
+        next_press_deadline = time.time() + 0.3
         with self.lock:
             while True:
-                self.lock.wait(0.450)
+                self.lock.wait(max(0, next_press_deadline - time.time()))
```

Comment:
The only place we call `notify_*` is keyup, so I guess that this is safe with Python's `Condition` implementation.